specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,327 @@
+  /// are also separate targets. Attaching to these targets allows interaction with them.
+  /// These attachments to the targets are sessions and are identified by sessionId.
+  /// To use this API, you should set `flatten` parameter to true when calling
+  /// `Target.attachToTarget` or `Target.setAutoAttach` `DevToolsProtocol` method.
+  /// Using `Target.setAutoAttach` is recommended as that would allow you to attach
+  /// to dedicated worker target, which is not discoverable via other APIs like
+  /// `Target.getTargets`.
+  /// For more information about targets and sessions, navigate to
+  /// \[DevTools Protocol Viewer\]\[GithubChromedevtoolsDevtoolsProtocolTotTarget\].
+  /// For more information about available methods, navigate to
+  /// \[DevTools Protocol Viewer\]\[GithubChromedevtoolsDevtoolsProtocolTot\]
+  /// The `sessionId` parameter is the sessionId for an attached target.
+  /// nullptr or empty string is treated as the session for the default target for the top page.
+  /// The `methodName` parameter is the full name of the method in the
+  /// `{domain}.{method}` format.  The `parametersAsJson` parameter is a JSON
+  /// formatted string containing the parameters for the corresponding method.
+  /// The `Invoke` method of the `handler` is run when the method
+  /// asynchronously completes.  `Invoke` is run with the return object of the
+  /// method as a JSON string.
+  ///
+  /// \[GithubChromedevtoolsDevtoolsProtocolTot\]: https://chromedevtools.github.io/devtools-protocol/tot "latest (tip-of-tree) protocol - Chrome DevTools Protocol | GitHub"
+  /// \[GithubChromedevtoolsDevtoolsProtocolTotTarget\]: https://chromedevtools.github.io/devtools-protocol/tot/Target "Chrome DevTools Protocol - Target domain"
+
+  HRESULT CallDevToolsProtocolMethodForSession(
+      [in] LPCWSTR sessionId,
+      [in] LPCWSTR methodName,
+      [in] LPCWSTR parametersAsJson,
+      [in] ICoreWebView2CallDevToolsProtocolMethodCompletedHandler* handler);
+}
+
+interface ICoreWebView2DevToolsProtocolEventReceivedEventArgs2 : IUnknown {
+
+  /// The sessionId of the target where the event originates from.
+  /// Empty string is returned as sessionId if the event comes from the default session for the top page.
+  /// \snippet ScriptComponent.cpp DevToolsProtocolEventReceivedSessionId
+  [propget] HRESULT SessionId([out, retval] LPWSTR* sessionId);
+}
+```
+
+## WinRT and .NET
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2
+    {
+        // ...
+        
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_10")]
+        {
+            // ICoreWebView2_10 members
+            // This is an overload for: public async Task<string> CallDevToolsProtocolMethodAsync(string methodName, string parametersAsJson);
+            public async Task<string> CallDevToolsProtocolMethodAsync(string sessionId, string methodName, string parametersAsJson);
```

Comment:
For overloaded methods the parameters shouldn't move between overloads.  So sessionId should go at the end. https://www.osgwiki.com/wiki/WinRT_API_Design_Guidelines#:~:text=AVOID%20being%20inconsistent%20in%20the%20ordering%20of%20parameters%20in%20overloaded%20members.